src/main/java/org/folio/rest/impl/PatronNoticePoliciesAPI.java

```diff
@@ -116,6 +119,51 @@ public void postPatronNoticePolicyStoragePatronNoticePolicies(
     });
   }
 
+  @Override
+  public void getPatronNoticePolicyStoragePatronNoticePoliciesByPatronNoticePolicyId(
+    String patronNoticePolicyId,
+    Map<String, String> okapiHeaders,
+    Handler<AsyncResult<Response>> asyncResultHandler,
+    Context vertxContext) {
+
+    vertxContext.runOnContext(v -> {
+      try {
+        String tenantId = okapiHeaders.get(RestVerticle.OKAPI_HEADER_TENANT);
+        PostgresClient pgClient = PostgresClient.getInstance(vertxContext.owner(), tenantId);
+
+        Criteria criteria = new Criteria()
+          .addField("'id'")
+          .setOperation(Criteria.OP_EQUAL)
+          .setValue(patronNoticePolicyId);
+
+        pgClient.get(PATRON_NOTICE_POLICY_TABLE, PatronNoticePolicy.class, new Criterion(criteria), false, get -> {
+          if (get.failed()) {
+            logger.error(get.cause());
+            asyncResultHandler.handle(Future.succeededFuture(
+              GetPatronNoticePolicyStoragePatronNoticePoliciesByPatronNoticePolicyIdResponse
+                .respond500WithTextPlain(get.cause())));
+            return;
+          }
+
+          if (get.result().getResults().isEmpty()) {
+            asyncResultHandler.handle(Future.succeededFuture(
+              GetPatronNoticePolicyStoragePatronNoticePoliciesByPatronNoticePolicyIdResponse
+                .respond404WithTextPlain(NOT_FOUND)));
+            return;
+          }
+
+          asyncResultHandler.handle(Future.succeededFuture(
+            GetPatronNoticePolicyStoragePatronNoticePoliciesByPatronNoticePolicyIdResponse
+              .respond200WithApplicationJson(get.result().getResults().get(0))));
+        });
+      } catch (Exception e) {
+        logger.error(e);
+        asyncResultHandler.handle(Future.succeededFuture(
+          PostPatronNoticePolicyStoragePatronNoticePoliciesResponse.respond500WithTextPlain(e)));
+      }
+    });
+  }
+
```

Comment:
Use

    public void getPatronNoticePolicyStoragePatronNoticePoliciesByPatronNoticePolicyId(
      String patronNoticePolicyId,
      Map<String, String> okapiHeaders,
      Handler<AsyncResult<Response>> asyncResultHandler,
      Context vertxContext) {

      PgUtil.getById(PATRON_NOTICE_POLICY_TABLE, PatronNoticePolicy.class, patronNoticePolicyId,
          okapiHeaders, vertxContext,
          GetPatronNoticePolicyStoragePatronNoticePoliciesByPatronNoticePolicyIdResponse.class,
          asyncResultHandler);
    }

domain-models-runtime has PgUtil.getById since version 23.3.0.

Reply:
Thanks. It looks much better now.

Reply:
Well done!